gluten-core/src/main/scala/io/glutenproject/utils/SelectiveExecution.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.glutenproject.utils
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.catalyst.plans.QueryPlan
+import org.apache.spark.sql.SparkSession
+import java.lang.Throwable
+import scala.util.Try
+import scala.collection.immutable.List
+
+object SelectiveExecution extends Logging{
+    var defaultEnabled = true
+    val CONF_OPTION = "gluten_enabled"
```

Comment:
there is a parameter called `spark.gluten.sql.enable.native.engine` to control whether to use gluten native engine, what's the difference ?

Reply:
@zzcclp  This parameter is only consulted in the pre and post columnar transitions. There is a lot of scanning the plan by Gluten and some transformations that seem happen in the other plugins, which are not turned off with this flag.
This proposed change is much more hermetic in turning off all of the impacts of Gluten on a specific DataFrame operation

Reply:
Sounds reasonable to me.

And it seems that it should be safe to remove the old `spark.gluten.sql.enable.native.engine` at the same time. Can you try?

Also please rename the new option. I would suggest `spark.gluten.enabled`.

Reply:
Is it OK merge? @zhztheplayer  can you remove the old spark.gluten.sql.enable.native.engine after this PR is merged?

Reply:
Maybe I'll rename the flag to `spark.gluten.sql.enable.native.engine`, so that wherever it's used in the tests, they'll still work.
How does that sound?

Reply:
Can you add a description to configuration.md? It's better to put an example how to use the feature to offload one SQL there. That will be very useful

Reply:
> Maybe I'll rename the flag to spark.gluten.sql.enable.native.engine

That's OK to me.